connector/connect/server/src/main/scala/org/apache/spark/sql/connect/ml/AlgorithmRegisty.scala

```diff
@@ -0,0 +1,157 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.connect.ml
+
+import org.apache.spark.connect.proto
+import org.apache.spark.ml
+import org.apache.spark.ml.{Estimator, Model}
+import org.apache.spark.ml.classification.TrainingSummary
+import org.apache.spark.ml.util.MLWriter
+import org.apache.spark.sql.DataFrame
+
+object AlgorithmRegistry {
+
+  def get(name: String): Algorithm = {
+    name match {
+      case "LogisticRegression" => new LogisticRegressionAlgorithm
+      case _ =>
+        throw new IllegalArgumentException()
+    }
+  }
+
+}
+
+abstract class Algorithm {
+
+  def initiateEstimator(uid: String): Estimator[_]
+
+  def getModelAttr(model: Model[_], name: String): Either[proto.MlCommandResponse, DataFrame]
+
+  def getModelSummaryAttr(
+      model: Model[_],
+      name: String,
+      datasetOpt: Option[DataFrame]): Either[proto.MlCommandResponse, DataFrame]
+
+  def loadModel(path: String): Model[_]
+
+  def loadEstimator(path: String): Estimator[_]
+
+  protected def getEstimatorWriter(estimator: Estimator[_]): MLWriter
+
+  protected def getModelWriter(model: Model[_]): MLWriter
+
+  def _save(
+      writer: MLWriter,
+      path: String,
+      overwrite: Boolean,
+      options: Map[String, String]): Unit = {
+    if (overwrite) {
+      writer.overwrite()
+    }
+    options.map { case (k, v) => writer.option(k, v) }
+    writer.save(path)
+  }
+
+  def saveModel(
+      model: Model[_],
+      path: String,
+      overwrite: Boolean,
+      options: Map[String, String]): Unit = {
+    _save(getModelWriter(model), path, overwrite, options)
+  }
+
+  def saveEstimator(
+      estimator: Estimator[_],
+      path: String,
+      overwrite: Boolean,
+      options: Map[String, String]): Unit = {
+    _save(getEstimatorWriter(estimator), path, overwrite, options)
+  }
+}
+
+class LogisticRegressionAlgorithm extends Algorithm {
```

Comment:
If we can use java reflection to invoke methods, we don't need the registry class, we just need some configuration data for registry.

If we plan to mandatorily enable spark connect mode since spark 4 for DBR, then we'd better use  java reflection invocation. Otherwise it is hard to support huge number of 3rd-party estimators.

CC @grundprinzip